src/ronin/profile/ProfileXComponents.sol

```diff
@@ -9,6 +9,29 @@ import "./ProfileHandler.sol";
 pragma solidity ^0.8.9;
 
 abstract contract ProfileXComponents is IProfile, ProfileHandler {
+  /**
+   * @inheritdoc IProfile
+   */
+  function execCreateRollup(address id, uint32 rollupId) external onlyContract(ContractType.ZK_ROLLUP_MANAGER) {
+    CandidateProfile storage _profile = _getId2ProfileHelper(id);
+
+    _requireNotOnRenunciation(id);
+
+    if (_registry[rollupId]) revert ErrRollupIdAlreadyRegistered(rollupId);
```

Comment:
If the rollup ID does not exist but accidentally matches any registered address, this tx will be reverted. Is this as intended?

Reply:
rollupId in range 2**32 so it cannot match

Reply:
I think this is a valid point, a candidate can register an address as 0x1 on purpose to block the rollup flow then change that address later.

Reply:
ack.